src/middlewares/productExists.js

```diff
@@ -0,0 +1,37 @@
+
+
+/* eslint-disable require-jsdoc */
+import { Product } from "../database/models";
+
+export async function IsProductExist(req, res, next) {
+  try {
+    const productId = req.params.id || req.body.product_id;
+    const product = await Product.findOne({ where: { id: productId } });
+    if (!product || product == null) {
+      return res
+        .status(404)
```

Comment:
- redudant status codes in responses make sure you correct it.
- ` return res
        .status(404).json({ status: "fail", message: "product does not exist" });`